imap_processing/codice/tests/test_decom.py

```diff
@@ -94,3 +221,21 @@ def test_ways_to_get_data(decom_test_data):
     data_value_using_list = decom_test_data[0][1]
     # Check if data is same
     assert data_value_using_key == data_value_using_list
+
+
+def test_unit_conversion(validation_data, filtered_analog):
+    """Test if unit conversion is correct"""
+    # Call the function to perform unit conversion on the validation data
+    engineering_data = validate_unit_conversion(validation_data, filtered_analog)
+
+    # Assert that both DataFrames have the same shape
+    assert engineering_data.shape == validation_data.shape
+
+    # Define a tolerance for floating-point precision
+    tolerance = 1e-6  # Adjust this tolerance as needed
+
+    # Loop through each column and each row to compare the values
+    for column in validation_data.columns:
+        for index, row in validation_data.iterrows():
+            # Compare each value in the column
+            assert abs(engineering_data.at[index, column] - row[column]) < tolerance
```

Comment:
Can you compare the arrays directly without needing to loop over the quantities?

`np.testing.assert_allclose(engineering_data, validation_data, rtol=1e-6)`